runtime/tests/org.eclipse.core.tests.harness/src/org/eclipse/core/tests/harness/FileSystemHelper.java

```diff
@@ -81,4 +98,92 @@ public static void clear(java.io.File file) {
 			CoreTest.log(CoreTest.PI_HARNESS, new Status(IStatus.WARNING, CoreTest.PI_HARNESS, IStatus.OK, message, null));
 		}
 	}
+
+	/**
+	 * Creates a symbolic link. Should only be called on platforms where symbolic
+	 * links can actually be created, i.e. an "ln" command is available.
+	 *
+	 * @param basedir    folder in which the symbolic link should be created
+	 * @param linkName   name of the symbolic link
+	 * @param linkTarget target to which the symbolic link should point
+	 * @param isDir      <code>true</code> if the link should point to a folder
+	 * @throws IllegalStateException in case symlink creation is not supported
+	 * @throws IOException           in case the creation of the symlink failed for
+	 *                               other reasons
+	 */
+	public static void createSymLink(File basedir, String linkName, String linkTarget, boolean isDir)
+			throws IOException {
+		// The following code creates even a link if
+		// Files.createSymbolicLink(new File(basedir, linkName).toPath(), new
+		// File(basedir, linkTarget).toPath());
+		// would throw java.nio.file.FileSystemException "missing rights"
```

Comment:
@HeikoKlare can you tell if this is a common scenario (for tests?).
When looking at the code my fist thought was if Files.createSymbolicLink() can be used.

Reply:
@HannesWell Unfortunately, I do not know whether this it is (still) a relevant/common scenario. I have only moved/refactored this code from the `CoreTest` class without challenging the necessity and way it is implemented.